app/src/main/java/com/github/geohunt/app/data/repository/AppContainer.kt

```diff
@@ -15,40 +19,42 @@ import java.lang.IllegalStateException
  * [Hilt](https://developer.android.com/training/dependency-injection/hilt-android) is added to
  * the codebase.
  */
-class AppContainer private constructor(application: Application) {
-    init {
-        try {
-            Firebase.database.setPersistenceEnabled(true)
-        } catch(_: DatabaseException) { /* This is already the case, why forcing it */ }
-    }
-
+class AppContainer private constructor(dbInstance: FirebaseDatabase, storageInstance: FirebaseStorage, application: Application) {
     val database = Firebase.database
-    val storage = Firebase.storage
+    val location: LocationRepository = LocationRepository(
+        SharedLocationManager(application.applicationContext)
+    )
 
+    val image = ImageRepository(storageInstance)
     val auth = AuthRepository()
+    val user = UserRepository(image, auth, dbInstance)
 
-    val image = ImageRepository()
-    val user = UserRepository(image, auth)
-    val follow = FollowRepository(auth)
-
-    val challenges = ChallengeRepository(user, image, auth)
-    val activeHunts = ActiveHuntsRepository(auth)
+    val challenges = ChallengeRepository(user, image, auth, dbInstance)
+    val activeHunts = ActiveHuntsRepository(auth, dbInstance)
+    val claims = ClaimRepository(auth, image, dbInstance)
+    val follow = FollowRepository(auth, dbInstance)
 
     val feedUseCase = GetUserFeedUseCase(auth, challenges, follow)
 
     companion object {
         private var container: AppContainer? = null
 
-        /**
-         * Returns the singleton instance of [AppContainer].
-         */
-        fun getInstance(application: Application): AppContainer {
+        private fun getInstance(dbInstance: () -> FirebaseDatabase, storage: () -> FirebaseStorage, application: Application) : AppContainer {
             if (container == null) {
-                container = AppContainer(application)
+                container = AppContainer(dbInstance(), storage(), application)
             }
             return container as AppContainer
         }
 
+        /**
+         * Get the instance for the current AppContainer
+         */
+        fun getInstance(application: Application): AppContainer = getInstance(
+            { FirebaseDatabase.getInstance().apply { this.setPersistenceEnabled(true) } },
```

Comment:
You should check for exceptions, as `this.setPersistenceEnabled(true)` might have already been called.
Something similar to this:
```suggestion
            { FirebaseDatabase.getInstance().apply { 
                try {
                    this.setPersistenceEnabled(true) 
                } catch(_: Exception) {}
            } },
```